mizar/common/rpc.py

```diff
@@ -98,24 +98,32 @@ def get_substrate_ep_json(self, ip, mac):
         jsonconf = json.dumps(jsonconf)
         return jsonconf
 
-    def update_substrate_ep(self, ip, mac):
+    def update_substrate_ep(self, ip, mac, task):
         jsonconf = self.get_substrate_ep_json(ip, mac)
         cmd = f'''{self.trn_cli_update_ep} \'{jsonconf}\''''
         logger.info("update_substrate_ep: {}".format(cmd))
         returncode, text = run_cmd(cmd)
-        logger.info("returns {} {}".format(returncode, text))
+        logger.info("update_subpstrate_ep {} returns {} {}".format(
+            cmd, text, returncode))
+        if (CONSTANTS.RPC_ERROR_CODE in text or
+            CONSTANTS.RPC_FAILED_CODE in text or
+                CONSTANTS.RPC_FATAL_CODE in text):
+            task.raise_temporary_error(
+                "Update_agent_substrate ep returned ERROR! Retrying as agent may have not yet been loaded.")
```

Comment:
I believe you mean to write "update_substrate_ep" instead of "Update_agent_substrate ep" here.

Reply:
fixed. thanks for the catch.